File: Usuario/backend/src/main/java/com/signspeak/backend/dto/CrearConversacionRequest.java
package com.signspeak.backend.dto;

import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Map;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class CrearConversacionRequest {

    @NotNull(message = "El ID de usuario es obligatorio")
    private Long idUsuario;

    @Size(max = 200, message = "El título no puede superar los 200 caracteres")
    private String tituloConversacion;  // ✅ CAMBIADO de "titulo" a "tituloConversacion"

    private Map<String, Object> metadata;
}

File: Usuario/mobile/app/src/main/java/com/signspeak/data/remote/dto/conversacion/CrearConversacionRequest.kt
package com.signspeak.data.remote.dto.conversacion

import com.google.gson.annotations.SerializedName

data class CrearConversacionRequest(
    val idUsuario: Long,
    val tituloConversacion: String? = null,
    @SerializedName("metadata")
    val metadata: Map<String, Any?>? = null
)